src/main/java/htsjdk/samtools/reference/FastaSequenceFile.java

```diff
@@ -59,6 +60,16 @@ public FastaSequenceFile(final Path path, final boolean truncateNamesAtWhitespac
         this.in = new FastLineReader(IOUtil.openFileForReading(path));
     }
 
+    /**
+     * Constructs a FastaSequenceFile that reads from the specified input stream (which must not be compressed, i.e.
+     * the caller is responsible for decompressing the stream).
+     */
+    public FastaSequenceFile(final InputStream inputStream, final boolean truncateNamesAtWhitespace) {
```

Comment:
`reset()` is going to fail in a confusing way.  Since this isn't seekable it's not possible to modify it to work with this.  We should either add a better message when calling reset on a stream created this way, or require this to be seekable and modify reset.

Reply:
Changed to use `SeekableStream` so `reset()` can be implemented properly.